src/services/game-process-service.js

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios';
+import chalk from 'chalk';
+
+export class GameProcessService {
+    constructor({ gameService, gameSourcesService, pubSubQueueProvider }) {
+        this.paused = false;
+        this.gameService = gameService;
+        this.gameSourceService = gameSourcesService;
+        this.queue = pubSubQueueProvider;
+        this.gamesQueueName = process.env.GAMES_QUEUE || 'games';
+        const host = process.env.HTML_SCRAPER_SERVICE_HOST || 'localhost';
+        const port = process.env.HTML_SCRAPER_SERVICE_PORT || 3002;
+        this.scraperBaseUrl = `http://${host}:${port}/api/`;
+    }
+
+    async process() {
+        while (!this.paused) {
+            console.log(`Checking ${this.gamesQueueName} queue for game to process...`);
+            await this.getQueuedGame();
```

Comment:
This should only ever be called once. ch.consume() binds that function to be called any time it sees a new entry in the queue, so all you're doing is either creating an infinite amount of bindings which would eventually fill up memory or replacing the previous bind with the exact same new one.

Reply:
Ahh I see. Cool I'll fix that.